.github/workflows/weekly-pulumi-update.yml

```diff
@@ -42,13 +44,12 @@ jobs:
 
         git checkout -b update-pulumi/${{ github.run_id }}-${{ github.run_number }}
 
-        go get github.com/pulumi/pulumi/pkg/v3
+        PULUMI_VERSION=$(pulumi version)
 
-        go get github.com/pulumi/pulumi/sdk/v3
+        echo "Update Pulumi dependencies to $PULUMI_VERSION"
+        go run scripts/build.go update-pulumi-deps "$PULUMI_VERSION"
```

Comment:
Instead of updating dependencies in the top-level Go module, this will update *every* Go module in the repo.